### PR TITLE
add [cancel_action], implements #1427

### PR DESCRIPTION
**Pull request by gfgtdf**
this adds a new tag [cancel_action], it currently only has an effect in
move actions. It can be used in enter_hex/exit_hex events to abort the
current movement, It can also be used in moveto events there is just
cancels possible following attacks if the user issued a move+attack
action.

This also changes the behaviour of moveto and enter/exit_hex event so
that they now no longer cancel the movement automatically, also
[allow_undo] no longer has an effect on whether the move is cancelled,
it only allows undoing of the move.

### CODE REVIEW

**Comment by CelticMinstrel** [comment]
> It can also be used in moveto events there is just
cancels possible following attacks if the user issued a move+attack
action.

Uh, I'm pretty sure the event firing cancels the attack already.

**Comment by gfgtdf** [comment]
> Uh, I'm pretty sure the event firing cancels the attack already.

This pr changes that, see #1427 for discussion.

> EDIT: Also, shouldn't it also be usable in any event that would be triggered by a move? For example, village capture.


Yes it is.

**Comment by GregoryLundberg** [comment]
I take it these are submissions for the lamest commit comment on GitHub?

**Comment by gfgtdf** [comment]
I'll squash them before i merge (or i'll porbably just use sqash-and-merge), they contain just fixups.

File: data/core/macros/event-utils.cfg [hunk 1]
```diff
@@ -42,9 +42,5 @@
 #define NO_INTERRUPT_NO_UNDO
     # This can be used in mid-movement events (enter_hex and exit_hex) to cause
     # no interruption of movement, yet prevent the move from being undone.
-    [allow_undo]
-    [/allow_undo]
-    [event]
-        name=moveto
-    [/event]
+	# TODO: maybe a deprecation message?
```
Comment (CelticMinstrel) [inline data/core/macros/event-utils.cfg]: Since this is ActionWML, you can use the `[deprecated_message]` tag.

**Comment by gfgtdf** [comment]
> please avoid spurious changes such as all those which (appear?) to move the comma from one line to another. The practice is debatable to begin with. It makes it quite hard to see what's changed and what is ignorable noise.

You can just look at the single commits then (in factit is is just one that really changes the c++ code).

**Pull request merged**